exist-core/src/main/java/org/exist/xquery/functions/fn/FunUnparsedText.java

```diff
@@ -209,6 +209,10 @@ private Source getSource(final String uriParam) throws XPathException {
                 throw new XPathException(this, ErrorCodes.FOUT1170, "Could not find source for: " + uriParam);
             }
 
+            if (source.getInputStream()==null) {
```

Comment:
I don't think this is safe to do. In some instances calling `source.getInputStream` is not an idempotent operation.

Reply:
ahhhhhh good catch

Reply:
reworked